metricbeat/docs/modules/openai.asciidoc

```diff
@@ -0,0 +1,74 @@
+////
+This file is generated! See scripts/mage/docs_collector.go
+////
+
+:modulename: openai
+:edit_url: https://github.com/elastic/beats/edit/main/x-pack/metricbeat/module/openai/_meta/docs.asciidoc
+
+
+[[metricbeat-module-openai]]
+[role="xpack"]
+== openai module
+
+beta[]
+
+This is the openai module.
+
+
+
+:edit_url:
+
+[float]
+=== Example configuration
+
+The openai module supports the standard configuration options that are described
+in <<configuration-metricbeat>>. Here is an example configuration:
+
+[source,yaml]
+----
+metricbeat.modules:
+- module: openai
+  metricsets: ["usage"]
+  enabled: false
+  period: 1h
+
+  # # Project API Keys - Multiple API keys can be specified for different projects
+  # api_keys:
+  # - key: "api_key1"
+  # - key: "api_key2"
+
+  # # API Configuration
+  # ## Base URL for the OpenAI usage API endpoint
+  # api_url: "https://api.openai.com/v1/usage"
+  # ## Custom headers to be included in API requests
+  # headers:
+  # - "k1: v1"
+  # - "k2: v2"
+  ## Rate Limiting Configuration
+  # rate_limit:
+  #   limit: 12 # seconds between requests
+  #   burst: 1  # max concurrent requests
+  # ## Request Timeout Duration
+  # timeout: 30s
+
+  # # Data Collection Configuration
+  # collection:
+  #   ## Number of days to look back when collecting usage data
+  #   lookback_days: 30
+  #   ## Whether to collect usage data in realtime. Defaults to false as how
+  #   # OpenAI usage data is collected will end up adding duplicate data to ES
+  #   # and also making it harder to do analytics. Best approach is to avoid
+  #   # realtime collection and collect only upto last day (in UTC). So, there's
+  #   # at most 24h delay.
+  #   realtime: false----
```

Comment:
This PR is failing docs ci at least in part due to this code block never being closed. The closing `----` needs to be on a newline for it to work. I know this is a generated file, but I'll add a suggestion to illustrate this point:

```suggestion
  #   realtime: false
----
```

Reply:
Oh, thanks for this!****